### PR TITLE
chore: add RFC document for avatar-upload

### PR DESCRIPTION
**Pull request by ozgurkececioglu**
[FX-2889]

### Description

I created an RFC document according to the requirements of the ticket. Any questions/suggestions will be welcomed.

### How to test

No need testing

### Screenshots

None.

### Development checks

- Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- Annotate all `props` in component with documentation
- Create `examples` for component
- Ensure that deployed demo has expected results and good examples
- Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- Self reviewed
- Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2889]: https://toptal-core.atlassian.net/browse/FX-2889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/decisions/09-avatar-upload.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Introducing upload functionality for avatar
+
+## Problem
+
+There is a new feature to get avatar uploads through `Avatar` component. So the component will look like an avatar but after hovering, it will show file upload mechanism to the user.
```
Comment (OleksandrNechai) [inline docs/decisions/09-avatar-upload.md:5]: After hovering?

Comment (ozgurkececioglu) [reply docs/decisions/09-avatar-upload.md:5]: I recheck everywhere and there is no mention of `hovering` 🤔 how did I misunderstand this way 😄 thanks for pointing out.

**Review by OleksandrNechai — APPROVED** [review]
Seems reasonable to me 👍

File: docs/decisions/09-avatar-upload.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Problem
+
+There is a new feature to help users upload their avatars. So the component will have the shape of an avatar but it will act like a file input. If the input has a selected or uploaded image, the image will be displayed in the background.
+
+### Requirements
+
+- It has all states like a FileInput
+- It can also act like a field so that if the field is empty, the user should be notified after form submission.
+- It is also expected to show warning and error according to the file that which user is trying to upload.
+- Drag-and-drop is not required but it is nice to have
```
Comment (dmaklygin) [inline docs/decisions/09-avatar-upload.md:12]: if it's not a mandatory, then it's not a requirement and we should remove this line :)

**Review by TomasSlama — COMMENTED** [review]
I'm just wondering if we the component should not have little bit bigger scope and look like this 
image

**Comment by ozgurkececioglu** [comment]
> I'm just wondering if we the component should not have little bit bigger scope and look like this image

I asked the same question to @MattVella and he said we can extend the component structure to include these parts too 👍 I will update the RFC according to that.

**Review by TomasSlama — APPROVED** [review]
👍

**Review by dmaklygin — APPROVED** [review]

**Pull request merged**